homeassistant/components/switcher_kis/switch.py

```diff
@@ -67,11 +69,30 @@ async def async_setup_entry(
     @callback
     def async_add_switch(coordinator: SwitcherDataUpdateCoordinator) -> None:
         """Add switch from Switcher device."""
+        entities: list[SwitchEntity] = []
+
         if coordinator.data.device_type.category == DeviceCategory.POWER_PLUG:
             async_add_entities([SwitcherPowerPlugSwitchEntity(coordinator)])
         elif coordinator.data.device_type.category == DeviceCategory.WATER_HEATER:
             async_add_entities([SwitcherWaterHeaterSwitchEntity(coordinator)])
 
+        if coordinator.data.device_type.category in (
+            DeviceCategory.SHUTTER,
+            DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT,
+            DeviceCategory.DUAL_SHUTTER_SINGLE_LIGHT,
+        ):
```

Comment:
This needs to be cleaned up so you call `async_add_entities` only once,

Reply:
it is done.